### PR TITLE
Adding shapes related to Address Book

### PR DESCRIPTION
**Pull request by megoth**
# New feature description

Adding shapes that relates to Address Books in Contacts.

# Checklist

- [x] All acceptance criteria are met.
- [x] Relevant documentation, if any, has been written/updated.
- [ ] The changelog has been updated, if applicable.
- [x] Commits in this PR are minimal and [have descriptive commit messages](https://chris.beams.io/posts/git-commit/).

### CODE REVIEW

**Comment by megoth** [comment]
A flaw with this approach is that the current shapes require target-classes added to the various triples (e.g. vcard:AddressBook in the group index, the people index, and the individual groups), something which currently isn't there.

But not sure how to use these shapes when the graph is split into multiple resources, like they are here.

**Comment by megoth** [comment]
A limitation that we haven't addressed in these files (and isn't addressed in the SHACL shape either) is that Data Browser requires the subject of vcard:AddressBook in index.ttl to have the id `this`. I don't think this is something that we can describe using shapes?

File: shapes/contacts/address-book.shex [hunk 1]
```diff
@@ -0,0 +1,11 @@
+PREFIX acl:   <http://www.w3.org/ns/auth/acl#>
+PREFIX dc:    <http://purl.org/dc/elements/1.1/>
+PREFIX vcard: <http://www.w3.org/2006/vcard/ns#>
+PREFIX xsd:   <http://www.w3.org/2001/XMLSchema#>
+
+vcard:AddressBook {
+  dc:title              xsd:string ;
```
Comment (podbrowser) [inline shapes/contacts/address-book.shex:7]: Enforce only one title:
`dc:title              xsd:string {1} ;`

File: shapes/contacts/address-book.shex [hunk 2]
```diff
@@ -0,0 +1,11 @@
...
+  vcard:nameEmailIndex  IRI ;
```
Comment (podbrowser) [inline shapes/contacts/address-book.shex:8]: `vcard:nameEmailIndex  IRI {1} ;`

File: shapes/contacts/address-book.shex [hunk 3]
```diff
@@ -0,0 +1,11 @@
...
+  vcard:groupIndex      IRI ;
```
Comment (podbrowser) [inline shapes/contacts/address-book.shex:9]: `vcard:groupIndex      IRI {1} ;`

Comment (kevin-inrupt) [inline shapes/contacts/address-book.shex:9]: `vcard:groupIndex      IRI  {1};`

File: shapes/contacts/address-book.shex (hunk 2 above)
[the remark this answers is not in the document]
Comment (megoth) [reply shapes/contacts/address-book.shex:8]: But {1,1} is the default cardinality? (At least that's what's said in page 23 in https://www.slideshare.net/jelabra/shex-by-example-97678930 )

[the remark this answers is not in the document]
Comment (megoth) [reply shapes/contacts/address-book.shex:8]: That said, it cost us little to be explicit, and if we can avoid confusion by doing it, that's a plus.

@VirginiaBalseiro let's add explicit cardinalities.

[the remark this answers is not in the document]
Comment (megoth) [reply shapes/contacts/address-book.shex:8]: We're adding explicit cardinalities in a new PR.

**Comment by megoth** [comment]
@VirginiaBalseiro will create a new PR with all of the work on contact shapes, so closing this.

**Comment by megoth** [comment]
Closed for https://github.com/inrupt/pod-browser/pull/68

**Pull request closed without merging**